src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -78,9 +82,17 @@ void ucc_tl_ucp_allgather_knomial_progress(ucc_coll_task_t *coll_task)
     ucc_status_t           status;
     size_t                 extra_count;
 
-    EXEC_TASK_TEST(UCC_KN_PHASE_INIT, "failed during ee task test",
-                   task->allgather_kn.etask);
+    uint32_t USE_CUDA = UCC_TL_UCP_TEAM_LIB(team)->cfg.allgather_use_cuda;
+
+    if(!USE_CUDA){
+        if (UCC_INPROGRESS == ucc_tl_ucp_test(task)){
```

Comment:
Would it be more concise to have`if(!USE_CUDA && UCC_INPROGRESS == ucc_tl_ucp_test(task))`?

Also, if `USE_CUDA` is a local variable it should be snake case by convention, `use_cuda`